File: Java/lab8/solutions/src/main/java/com/honolulu/Producer.java
package com.honolulu;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerConfig;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.apache.kafka.common.serialization.StringSerializer;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Properties;

public class Producer {
	private static final Logger log = LoggerFactory.getLogger(Producer.class);
	private final Properties props = new Properties();
	private KafkaProducer<String, String> producer;

	public Producer() {
		props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "http://localhost:9092,http://localhost:9093");
		props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
		props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
		producer = new KafkaProducer<>(props);
	}

	public void produceMessageToTopic(String topic) {

//		At this point you could send in AVRO the driver information to decouple
		ProducerRecord<String, String> producerRecord =
					new ProducerRecord<>(topic, "driver", "new_driver");

		producer.send(producerRecord);

		producer.flush();
		producer.close();

		log.info("Successfully produced messages");

		}

	public void killMessage(String topic) {

		ProducerRecord<String, String> producerRecord =
				new ProducerRecord<>(topic, null, "new_driver");

		producer.send(producerRecord);

		producer.flush();
		producer.close();

		log.info("Killed message");

	}
}

File: JS/lab8/solutions/producer.js
const { Kafka } = require('kafkajs')
var net = require('net');

const kafka = new Kafka({
  brokers: ['localhost:9092', 'localhost:9093'],
  clientId: 'lab8-producer',
})

const topic = 'new-driver' 
const producer = kafka.producer()


const sendMessage = () => {
  var messages = [{key: 'new_driver', value: 'new-driver'}];
  return producer.send({
      topic,
      messages: messages,
      acks: 0
  })
}

const run = async () => {
  await producer.connect()
  setInterval(sendMessage, 100)
}
run()